README.md

```diff
@@ -1,12 +1,238 @@
+
+To simplify the implementation of the sidecar container, you can use the [libsidecar](https://github.com/kubevirt/application-aware-quota-api/blob/main/libsidecar/sidecar.go) library. 
+To implement the sidecar container, you need to implement the following interface:
+
+```go
+type SidecarCalculator interface {
+    PodUsageFunc(podToEvaluate *corev1.Pod, existingPods []*corev1.Pod) (corev1.ResourceList, bool, error)
+}
+```
+
+The PodUsageFunc function takes as input the pod that needs to be evaluated and all the other pods in the cluster. 
+It returns the resources that should be counted for this pod, a boolean indicating if the pod belongs to an 
+application policy, and an error if something goes wrong during the evaluation.
+
+If a pod doesn't belong to a policy defined in the AAQ object, the policy won't be considered, or if an
+error has occurred, the policy won't be considered. In other words, the returned ResourceList is
+ignored.
+
+If a pod does not belong to any policy, native quota evaluation will apply, and pod counting will be
+the same as native ResourceQuota counting.
+
+If a pod belongs to multiple policies, the evaluation of all these policies will 
+apply to this pod, and the counting will be the sum of all policy evaluations.
+
+
+Please see [label-sidecar.go](https://github.com/kubevirt/application-aware-quota/blob/main/example_sidecars/label-sidecar/cmd/label-sidecar.go) for an example sidecar policy.
+for an example sidecar policy.
+
+### Configure AAQ Operator
+
+To configure AAQ Operator, set the following fields of the aaq object:
+
+* `aaq.spec.namespaceSelector` - determines in which namespaces AAQ's scheduling gate will be added to pods at creation time. This field follows the standard Kubernetes selector format.
+
+    ***Important Note***: if a namespace selector is not being defined, AAQ would target namespaces with the `application-aware-quota/enable-gating` label as default.
+
+    ***Warning***: Pods that set `.spec.nodeName` are not allowed to use in any namespace that AAQ is targeting. Such pods would fail at creation time. Therefore, users shouldn't be targeting system namespaces where `.spec.nodeName` might be set.
+    
+    For more information please see [this discussion](https://github.com/kubernetes/enhancements/issues/3521#issuecomment-2016957473)
+
+
+* `aaq.spec.configuration.allowApplicationAwareClusterResourceQuota` - (default = false) - set to true, to allow creation and management of ClusterAppsResourceQuota.
+
+  ***Note***: this setting might have some scheduling time cost where AAQ operates.
+
+
+* `aaq.spec.configuration.SidecarEvaluators` - Slice of standard Kubernetes container object. Adding containers to the 
+SidecarEvaluators will include sidecar calculator for custom counting. The containers should mount a directory under `/var/run/aaq-sockets` that's called `sockets-dir`.
+
```

Comment:
Maybe add a very small sentence about the sidecar container. maybe something like "this container will be using gRPC to evaluate the resources the pod is accounted for. The process of setting up the environment (e.g. gRPC client/server) is significantly simplified with the libsidecar library that handles the heavy lifting.

Reply:
Done